com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviourUpdater.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections.Generic;
+using Unity.Profiling;
+
+namespace MLAPI
+{
+    public class NetworkBehaviourUpdater
+    {
+        private HashSet<NetworkObject> m_Touched = new HashSet<NetworkObject>();
+
+        /// <summary>
+        /// Stores the network tick at the NetworkBehaviourUpdate time
+        /// This allows sending NetworkVariables not more often than once per network tick, regardless of the update rate
+        /// </summary>
+        public ushort CurrentTick { get; set; }
+
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+        private ProfilerMarker m_NetworkBehaviourUpdate = new ProfilerMarker($"{nameof(NetworkBehaviour)}.{nameof(NetworkBehaviourUpdate)}");
+#endif
+
+        internal void NetworkBehaviourUpdate(NetworkManager networkManager)
+        {
+            // Do not execute NetworkBehaviourUpdate more than once per network tick
+            ushort tick = networkManager.NetworkTickSystem.GetTick();
+            if (tick == CurrentTick)
+            {
+                return;
+            }
+
+            CurrentTick = tick;
+
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            m_NetworkBehaviourUpdate.Begin();
+#endif
+            try
+            {
```

Comment:
I'm a little confused and worried about this try-finally pattern wrapping. We bluntly suppress exceptions here. If we expect no exceptions, then there is no need for the try-finally pattern here — if we do expect, why suppress them here and do nothing about them?
(this might not be your code. if that's the case, ignore my question)

Reply:
That is indeed not my code. I copy pasted the content of the static NetworkBehaviourUpdate method here.
Tracking this down, looks like @mattwalsh-unity added this in commit 996d46dc4bf52c8a35e3ce4d960ebd297f7aff50 in NetworkedObject.cs when adding performance counters. I would guess it's to make sure even if there's an exception, we call "end()" on it, to make sure everything is tidied up?

Reply:
Also, in C# try-finally shouldn't suppress exceptions for this case. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/try-finally

Reply:
ugh, dodgy...